Metadata API changes (#2934)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-cloud-apis/metadata-api.md

---

> title: "Metadata API overview"

Reviewer: Should this be "About Metadata API"? I thought we weren't using "overview" but could very well be mistaken

Author: i did ask @runleonarun about this and she mentioned using 'overview' in the title was fine and that we weren't using was having an overview h2 as the intro of a page since the page would display a title followed by an h2. however i may have misunderstood. i can change this to 'about metadata api'

Author: changed this to 'about the metadata api'

id: "metadata-api"
---

> Every time dbt Cloud runs a project, it generates and stores information about the dbt project—in other words, metadata. It stores information on how it's executed, including accuracy, recency, configuration, and structure of the <Term id="view">views</Term> and tables in the warehouse.  

Reviewer: ```suggestion
Every time dbt Cloud runs a project, it generates and stores information about the dbt project — in other words, metadata. It stores information on how it's executed, including accuracy, recency, configuration, the structure of the <Term id="view">views</Term>, and tables in the warehouse.  
```

- [**Discovery**](/docs/dbt-cloud-apis/metadata-use-case-guides#discovery)* &mdash; Find and understand dbt assets to analyze in integrated tools using information like model and metric definitions, column info, and lineage. One example of this is the [dbt Semantic Layer integration](/guides/dbt-ecosystem/sl-partner-integration-guide). 
- [**Quality**](/docs/dbt-cloud-apis/metadata-use-case-guides#quality)* &mdash; Make sure users have correct and up-to-date data for their analyses by monitoring test failures, source freshness, run status, exposures, and dependencies.
- [**Operations**](/docs/dbt-cloud-apis/metadata-use-case-guides#operations)* &mdash;  Help data teams run dbt efficiently and reduce costs by using historical run data, including information like model build time and run counts.

*_Refer to [Using the Metadata API](/docs/dbt-cloud-apis/metadata-use-case-guides) for more info; additional detailed use case and integration guides are coming soon_.

## Product roadmap

> The 2023 Metadata API roadmap outlines three main uses: discovery, quality, and operation, with specific initiatives planned for each quarter:

Reviewer 2: The Metadata API roadmap includes initiatives to improve each of the API's main use cases: data discovery, data quality, and dbt operations.

<!--- tabs for discovery, quality, operations --->
<Tabs>

<TabItem value="discovery" label="Discovery">

> To improve discovery experiences, we’ll make it easier for API users to access the latest production state of a project. This is our primary focus in the first half of 2023. 

Reviewer 2: It's ok to leave time reference here

> -  Q1 &mdash; Rather than querying for each job or run, retrieve the project's logical state (definitions), execution results, and applied state (what exists in the database) from the most recent production runs in its environment.

Reviewer 2: Final rev i promise: 

Rather than querying for each job or run, retrieve the logical state (definitions), execution results, and applied state (what exists in the database) of each node in the project based on their most recent run in the production environment.

Author: no worries, thanks @cafzal !

> -  Q1 &mdash; Rather than querying for each job or run, retrieve the project's logical state (definitions), execution results, and applied state (what exists in the database) from the most recent production runs in its environment.

Author: ```suggestion
-  Q1 &mdash; Rather than querying for each job or run, retrieve the logical state (definitions), execution results, and applied state (what exists in the database) of each node in the project based on their most recent runs in the production environment.
```

> -  Q1 &mdash; Rather than querying for each job or run, retrieve the logical state (definitions), execution results, and applied state (what exists in the database) of each node in the project based on their most recent runs in the production environment.

Reviewer 2: Please remove the references to quarters for each of the line items

- Q2 &mdash; dbt v1.6: New and revised model and entity endpoints to unlock Semantic Layer use cases
- Q2 &mdash; Pagination for manageable responses and performance when querying long lists of dbt objects

> - Q2 &mdash; Instead of querying per job, get the DAG state based on the latest production runs of its environment.

Reviewer 2: @mirnawong1 Let's please expand a bit (text below) and move this one to the top. We're on track to release end of Q1. I'll share doc and use case updates in early April. 

"- Q1 &mdash; Instead of querying per job/run, get the project's logical state (definitions), execution results, and applied state (the output of execution - what exists in the database) based on the latest production runs of its environment."

- Q3 &mdash; Retrieve project information from the dbt logs, such as catalog information, without having to generate documentation.

</TabItem>

<TabItem value="quality" label="Quality">

> To improve customers’ data quality, we’re enhancing the ability for data teams to configure dbt and monitor for issues in dbt Cloud and integrated tools.  

Reviewer 2: This will be a key focus starting in 2024.

> - Q1 &mdash; Launch Metadata API support for dbt Core 1.5 elements of data contracts like model owners and constraints.

Reviewer 2: "- Q1 &mdash; Add Metadata API support for dbt Core 1.5 elements of data contracts like model owners, access, and constraints."

> - Q1 &mdash; Add Metadata API support for dbt Core 1.5 elements of data contracts like model owners, access, and constraints.

Reviewer 2: Let's remove quarter references

> - Q4 or later &mdash; Improvements to webhooks to support notifications about tests, freshness, and events per model rather than per run.

Reviewer 2: Remove since consolidated with next point

> - Q4 or later &mdash; Streaming results so users can access real-time metadata during a run.

Reviewer 2: "Streaming results so users can access real-time metadata during a run, such as notifications about specific events like model builds and test failures"

> - Q4 or later &mdash; Integrate with model freshness SLAs for intelligent scheduling.

Reviewer 2: Remove for now since depends on Orchestration

</TabItem>

<TabItem value="operations" label="Operations">

> To improve dbt operations, we’ll enable API users to access and query richer information about execution on dbt Cloud in more ergonomic ways. This is our primary focus in the second half of 2023. 

Reviewer 2: Ok to leave the year reference

> - Q3 &mdash; Enable easier queries across runs to analyze performance over time, such as time aggregation for execution results of individual models and the project as a whole.

Reviewer 2: Let's remove quarter references

- Q3 &mdash; Provide granular execution information from dbt Cloud, such as invocation history.

> - Q3 &mdash; Improvements to webhooks for notifications about run events.

Reviewer 2: Remove since more dependent on webhooks

> - Q4 &mdash; Streaming results to see the DAG as it builds during a run.

Reviewer 2: Add "[Later]" at the start of this to make it clear it isn't part of the other, 2023 projects

To improve discovery experiences, we’ll make it easier for API users to access the latest production state of a project. This is our primary focus in the first half of 2023. 

> - Q1 &mdash; dbt v1.5: Launch new model fields like owner, version, and constraints

Reviewer 2: Oh, smart moving this to Quality, We can delete from here since public models are covered as next step

To improve customers’ data quality, we’re enhancing the ability for data teams to configure dbt and monitor for issues in dbt Cloud and integrated tools.  

- Q1 &mdash; Launch Metadata API support for dbt Core 1.5 elements of data contracts like model owners and constraints.

> - Q4 &mdash; Improvements to webhooks to support notifications about tests, freshness, and events per model rather than per run.

Reviewer 2: Can we change these last 3 features to "Q4+"
